### ISSUE TITLE
 Stepfilled color cycle for background and edge different

### ISSUE THREAD

**Issue by Carreau** [matplotlib/matplotlib#6493]
I just realized that stepfilled (http://matplotlib.org/devdocs/examples/statistics/histogram_demo_multihist.html) 
has edge color cycle different from background color (which looks weird) didn't found any issues about that , so asking the question, as 2.x will change default style is is too late to change that ? (or is that a bug)

Matplotlib 1.5 and 2.0b

```
# %matplotlib inline

import numpy as np
import matplotlib.pyplot as plt

n_bins = 10
x = np.random.randn(1000, 3)

fig, ax = plt.subplots(nrows=1, ncols=1)
ax.hist(x, n_bins, histtype='step', stacked=True, fill=True)

plt.show()
```

Note that in attached example it's even weirder as the green as a gree edge, while the blue as a red edge and red has a blue edge.
![joz7rxkid5aaaaabjru5erkjggg](https://cloud.githubusercontent.com/assets/335567/15629362/077d765e-24cc-11e6-8fdb-ea2b351e50f3.png)

**Comment by tacaswell** [comment]
Going to go with that is just a bug, and old one at that http://matplotlib.org/1.4.0/examples/statistics/histogram_demo_multihist.html

![image](https://cloud.githubusercontent.com/assets/199813/15629827/ab834240-24f1-11e6-8a37-7b6890fb6637.png)

**Comment by tacaswell** [comment]
And this is not some strange overlay/aliasing bug, the colors really are different:

```python
In [23]: patches[0][0].get_edgecolor()
Out[26]: (0.12156862745098039, 0.4666666666666667, 0.7058823529411765, 1.0)

In [27]: patches[0][0].get_facecolor()
Out[27]: (0.17254901960784313, 0.6274509803921569, 0.17254901960784313, 1.0)
```

**Comment by tacaswell** [comment]
Very old: http://matplotlib.org/1.2.1/examples/pylab_examples/histogram_demo_extended.html (and that is as far back as we have built docs on the website)

![image](https://cloud.githubusercontent.com/assets/199813/15629906/ab020a92-24f4-11e6-8aa1-2c4cef4d1380.png)

**Comment by Carreau** [comment]
From a quick test I would say the edge colors and facecolors are just in reverse order of each other.

**Comment by Carreau** [comment]
quick investigating:

```
# axes/_axes.py Line 6170
            if fill:
                # add patches in reverse order so that when stacking,
                # items lower in the stack are plottted on top of
                # items higher in the stack

                ## Could change for reversed(color) here for consistency, but that's when filled = False, so does not really matter, except to keep the order of colors the same.
                for x, y, c in reversed(list(zip(xvals, yvals, color))):
                    patches.append(self.fill(
                        x, y,
                        closed=True,
                        facecolor=c))
            else:
                 ## change for reversed(color) here
                for x, y, c in reversed(list(zip(xvals, yvals, reversed(color)))):
                    split = 2 * len(bins)
                    patches.append(self.fill(
                        x[:split], y[:split],
                        closed=False, edgecolor=c,
                        fill=False))
```

**Comment by Carreau** [comment]
> QuLogic changed the title from Stepfilled color cycle for background and edge different to Stepfield color cycle for background and edge different? an hour ago

`Stepfilled` to `Stepfield` ? Is that autocorrect being overzealous as it was when I submitted the issue ? Or did I miss something ?

**Comment by QuLogic** [comment]
Sorry, I think that was waffle.io being buggy. I'm pretty sure I didn't change the title here.

**Comment by jenshnielsen** [comment]
Closed by #6494
